lib/workers/evm_server.rb

```diff
@@ -214,4 +262,43 @@ def start_workers
     @server.sync_workers
     @server.wait_for_started_workers
   end
+
+  ######################################################################
+  # Warning:
+  #
+  # The following methods can lead to unexpected (and likely unpleasant)
+  # behavior if used out of the scope of the orchestrator process.
+  #
+  # They change the global state which is used to determine the current
+  # server's identity. This intentionally will alter the values of calls
+  # such as MiqServer.my_server and MiqServer.my_guid, and also the
+  # contents of the global ::Settings constant.
+  ######################################################################
+  def for_each_server
+    initial_server = @server
+    servers_to_monitor.each do |s|
+      impersonate_server(s)
+      yield
+    end
+  ensure
+    clear_server_caches if @server != initial_server
+  end
+
+  def impersonate_server(s)
+    return if s == @server
```

Comment:
I think this would be clearer named `@current_server` or something to that effect.